src/main/java/roomescape/exception/reservation/DuplicatedReservationException.java

```diff
@@ -0,0 +1,10 @@
+package roomescape.exception.reservation;
+
+import org.springframework.http.HttpStatus;
+import roomescape.exception.CustomException;
+
+public class DuplicatedReservationException extends CustomException {
+    public DuplicatedReservationException() {
+        super("중복된 예약입니다.", HttpStatus.CONFLICT);
```

Comment:
409 status code에 대해 조금더 알아보시면 좋겠네요

Reply:
1. HTTP 409 Conflict 응답 상태 코드는 서버의 현재 상태와 요청이 충돌했음을 나타냅니다.
    - 현재와 같이 예약이 이미 잡힌 시간대여서 예약이 불가능할 때 409는 적절하다 생각합니다!
2. 응답은 충돌에 대해 설명하는 본문을 포함해야 합니다. 
    - 충돌에 대해 설명하는 본문이 조금 부족하게 느껴질 수도 있는 거 같아 수정해주었습니다.

Reply:
> The 409 (Conflict) status code indicates that the request could not be completed due to a conflict with the current state of the target resource. This code is used in situations where the user might be able to resolve the conflict and resubmit the request.

> Conflicts are most likely to occur in response to a PUT request. For example, if versioning were being used and the representation being PUT included changes to a resource that conflict with those made by an earlier (third-party) request, the origin server might use a 409 response to indicate that it can't complete the request. In this case, the response representation would likely contain information useful for merging the differences based on the revision history

duplicate되었다고 해서 conflict를 사용하는것 같진 않네요. 🤔

Reply:
"상태 충돌" 이라고 이해하면 409를 많이 사용하게 될 것 같네요.

> This code is used in situations where the user might be able to resolve the conflict and resubmit the request

조금더 살펴보면, 유저가 재시도하면 해결할 수 있는 상황에 쓰인다고 하네요. duplicate는 유저가 재시도하면 해결될까요?

> For example, if versioning were being used and the representation being PUT included changes to a resource that conflict with those made by an earlier (third-party) request, the origin server might use a 409 response to indicate that it can't complete the request.

어떤 요청을 받고 처리가 **완료되기 전에** 다른 요청을 받은 경우 409를 내릴수 있다고 하네요.

즉 409는 동시성 concurrency에 관한 에러라고 이해하면 좋을것 같아요.

Reply:
이 내용에 관해서도 많은 토론이 있었고 진행중이기도 하네요. https://stackoverflow.com/questions/3290182/which-status-code-should-i-use-for-failed-validations-or-invalid-duplicates

시간이 있다면 여러 의견들과 그 근거들을 읽어보는것도 좋을것 같아요.